src/app/directives/scroll.directive.spec.ts

```diff
@@ -1,10 +1,31 @@
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { Component, DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
 import { ScrollDirective } from './scroll.directive';
-import { ElementRef } from '@angular/core';
+
+@Component({
+  template: `<div id="tableTop"></div><button appScroll>`
+})
+class TestScrollComponent {}
 
 describe('ScrollDirective', () => {
-  const element = new ElementRef('button');
-  it('should create an instance', () => {
-    const directive = new ScrollDirective(element);
-    expect(directive).toBeTruthy();
+  let component: TestScrollComponent;
+  let fixture: ComponentFixture<TestScrollComponent>;
+  let inputEl: DebugElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [ScrollDirective, TestScrollComponent]
+    });
+
+    fixture = TestBed.createComponent(TestScrollComponent);
+    component = fixture.componentInstance;
+    inputEl = fixture.debugElement.query(By.css('button'));
+  });
+
+  it('should scroll to the top of the page', () => {
```

Comment:
Ooh fancy.  Can we add a line here to initially scroll the page down some amount first? Otherwise we don't really know if the scroll to top worked

Reply:
Done